torchaudio/csrc/cuctc/src/bitonic_topk/pow2_utils.cuh

```diff
@@ -0,0 +1,173 @@
+/**
+ *  Modified from Rapidsai/raft(https://github.com/rapidsai/raft)
+ * 
+ */
+
+#pragma once
+#include <type_traits>
+
+namespace cu_ctc
+{
+    /**
+     * @brief Give logarithm of the number to base-2
+     * @tparam IntType data type (checked only for integers)
+     */
+    template <typename IntType>
+    constexpr __device__ __forceinline__ IntType log2(IntType num, IntType ret = IntType(0))
```

Comment:
@yuekaizhang `clang++` does not seem to support recursive inline, thus it fails to compile. 

```
fbcode/pytorch/audio/torchaudio/csrc/cuctc/src/bitonic_topk/pow2_utils.cuh: In function ‘constexpr int cu_ctc::topk::calc_capacity(int)’:
fbcode/pytorch/audio/torchaudio/csrc/cuctc/src/bitonic_topk/pow2_utils.cuh:16:1: error: inlining failed in call to ‘always_inline’ ‘constexpr IntType cu_ctc::log2(IntType, IntType) [with IntType = int]’: function not considered for inlining
   16 |     constexpr __device__ __forceinline__ IntType log2(IntType num, IntType ret = IntType(0))
      | ^   
fbcode/pytorch/audio/torchaudio/csrc/cuctc/src/bitonic_topk/pow2_utils.cuh:18:42: note: called from here
   18 |         return num <= IntType(1) ? ret : log2(num >> IntType(1), ++ret);
      |                                      ~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Looking at the original implementation, it was a regular inline, but perhaps you changed this to `__forceinline__`. Is this change significant for the performance? If not I am thinking to revert this to regular inline for `clang++`.

https://github.com/rapidsai/raft/blob/83c326ec42fbe64ad5149a1a12a3c754a88c5c71/cpp/include/raft/util/cuda_dev_essentials.cuh#L74-L78

Reply:
Sure. I will remove this __forceinline__.